pkg/database/database.go

```diff
@@ -62,19 +37,33 @@ type DBInterface interface {
 }
 
 type Database struct {
-	db          *sqlx.DB
-	info        DBInfoInterface
-	paramParser DBParamParser
+	DB       *sqlx.DB
+	Flavor   sqlbuilder.Flavor
+	Selector facade.DBSelector
+	Filter   facade.DBFilter
+	Sorter   facade.DBSorter
+	Inserter facade.DBInserter
+	Deleter  facade.DBDeleter
 }
 
 func NewDatabase() (DBInterface, error) {
 	env, err := newDatabaseEnvironment()
 	if err != nil {
 		return nil, err
 	}
+
+	var creator facade.DBCreator
+	if c, ok := RegisteredDBCreators[env[DbKindEnvVar]]; ok {
+		creator = c(env)
+	} else {
+		panic(fmt.Sprintf("No database registered with name %s", env[DbKindEnvVar]))
+	}
+
+	conn := establishConnection(creator.GetDriverName(), creator.GetConnectionString())
+
 	var database DBInterface
-	if f, ok := RegisteredDatabases[env[DbKindEnvVar]]; ok {
-		database = f(env)
+	if init, ok := RegisteredDatabases[env[DbKindEnvVar]]; ok {
+		database = init(conn)
 	} else {
 		panic(fmt.Sprintf("No database registered with name %s", env[DbKindEnvVar]))
```

Comment:
No changes here, but noticed the other part approach. Golang is about returning errors, so panicking here does not look something we want to do.